plugins/license-fact-providers/scancode/src/main/kotlin/ScanCodeLicenseFactProvider.kt

```diff
@@ -0,0 +1,131 @@
+import org.ossreviewtoolkit.plugins.api.PluginDescriptor
+import org.ossreviewtoolkit.plugins.licensefactproviders.api.LicenseFactProvider
+import org.ossreviewtoolkit.plugins.licensefactproviders.api.LicenseFactProviderFactory
+import org.ossreviewtoolkit.utils.common.Os
+import org.ossreviewtoolkit.utils.common.realFile
+
+private const val FALLBACK_DIR = "/opt/scancode-license-data"
+
+/** The configuration for the ScanCode license fact provider. */
+data class ScanCodeLicenseFactProviderConfig(
+    /**
+     * The directory that contains the ScanCode license texts. If not set, the provider will try to locate the ScanCode
+     * license text directory using a heuristic based on the path of the ScanCode binary.
+     */
+    val scanCodeLicenseTextDir: String?
+)
+
+@OrtPlugin(
+    id = "ScanCode",
+    displayName = "ScanCode License Fact Provider",
+    description = "A license fact provider that reads license information from a local ScanCode installation.",
+    factory = LicenseFactProviderFactory::class
+)
+class ScanCodeLicenseFactProvider(
+    override val descriptor: PluginDescriptor = ScanCodeLicenseFactProviderFactory.descriptor,
+    private val config: ScanCodeLicenseFactProviderConfig
+) : LicenseFactProvider {
+    /**
+     * The directory that contains the ScanCode license texts. This is located using a heuristic based on the path of
+     * the ScanCode binary.
+     */
+    private val scanCodeLicenseTextDir: File by lazy {
+        if (config.scanCodeLicenseTextDir != null) {
+            File(config.scanCodeLicenseTextDir).takeIf { it.isDirectory }.also {
+                requireNotNull(it) {
+                    "Configured ScanCode license text directory '${config.scanCodeLicenseTextDir}' does not exist or " +
+                        "is not a directory."
+                }
+
+                logger.debug { "Using configured ScanCode license text directory: ${it.absolutePath}" }
+                return@lazy it
+            }
+        }
+
+        logger.debug { "Trying to locate the ScanCode license text directory..." }
+
+        val scanCodeExeDir = Os.getPathFromEnvironment("scancode")?.realFile?.parentFile
+
+        if (scanCodeExeDir == null) {
+            logger.debug { "Could not locate the ScanCode executable directory." }
+        } else {
+            logger.debug { "Located ScanCode executable directory: ${scanCodeExeDir.absolutePath}" }
+        }
+
+        val pythonBinDir = listOf("bin", "Scripts")
+        val scanCodeBaseDir = scanCodeExeDir?.takeUnless { it.name in pythonBinDir } ?: scanCodeExeDir?.parentFile
+
+        if (scanCodeBaseDir == null) {
+            logger.debug { "Could not locate the ScanCode base directory." }
+        } else {
+            logger.debug { "Located ScanCode base directory: ${scanCodeBaseDir.absolutePath}" }
+        }
+
+        val licenseDir =
+            scanCodeBaseDir?.walkTopDown()?.find { it.isDirectory && it.endsWith("licensedcode/data/licenses") }
```

Comment:
You could just use `walk()` as `FileWalkDirection.TOP_DOWN` is the default, which makes this line fit into one with the above.